compose_generator/protocols.py

```diff
@@ -0,0 +1,44 @@
+from dataclasses import dataclass
+from typing import Protocol
+
+
+@dataclass(frozen=True)
```

Comment:
The protocols doesn't need to have the dataclass decorator

Reply:
done